src/tainting/Taint.ml

```diff
@@ -84,11 +85,32 @@ let length_of_call_trace ct =
 
 type sink = { pm : Pattern_match.t; rule_sink : R.taint_sink } [@@deriving show]
 
+let compare_metavar_env env1 env2 =
+  (* It's important that we only return 0 if the two bindings are
+     structurally equal. Otherwise, there will be many duplicates.
+     We use Stdlib.compare in the other case because deriving
+     compare is rather difficult and the specific ordering doesn't
+     matter. *)
+  if Metavariable.Structural.equal_bindings env1 env2 then 0
+  else Stdlib.compare env1 env2
+
+let compare_matches pm1 pm2 =
+  let compare_rule_id =
```

Comment:
Would be good to follow the same style as in the other `compare` functions:
```ocaml
match ... with
| 0 -> ...
| other -> other
```
Unless you have a point to prefer this other style? I used to write them the same way, but I very recently changed to the `match ... with | 0 -> ...` as it feels more readable. (I believe I copied it from Brandon.)